datastore/db/bytebase/FertiScan/update_inspection_function.sql

```diff
@@ -0,0 +1,541 @@
+    json_inspection_id := (p_input_json->>'inspection_id')::uuid;
+    IF json_inspection_id IS NOT NULL AND json_inspection_id != inspection_id THEN
+        RAISE EXCEPTION 'Inspection ID mismatch: % vs %', inspection_id, json_inspection_id;
+    END IF;
+
+    -- Check if the provided inspector_id matches the existing one
+    SELECT inspector_id INTO existing_inspector_id
+    FROM "fertiscan_0.0.11".inspection
+    WHERE id = p_inspection_id;
+    IF existing_inspector_id IS NULL OR existing_inspector_id != p_inspector_id THEN
+        RAISE EXCEPTION 'Unauthorized: Inspector ID mismatch or inspection not found';
+    END IF;
+
+    -- Upsert company information and get the ID
+    company_info_id := "fertiscan_0.0.11".upsert_organization_info(p_input_json->'company');
+    updated_json := jsonb_set(updated_json, '{company,id}', to_jsonb(company_info_id));
+
+    -- Upsert manufacturer information and get the ID
+    manufacturer_info_id := "fertiscan_0.0.11".upsert_organization_info(p_input_json->'manufacturer');
+    updated_json := jsonb_set(updated_json, '{manufacturer,id}', to_jsonb(manufacturer_info_id));
+
+    -- Upsert label information and get the ID
+    label_info_id := "fertiscan_0.0.11".upsert_label_information(
+        p_input_json->'product',
+        company_info_id,
+        manufacturer_info_id
+    );
+    updated_json := jsonb_set(updated_json, '{product,id}', to_jsonb(label_info_id));
+
+    -- Update metrics related to the label
+    PERFORM "fertiscan_0.0.11".update_metrics(label_info_id, p_input_json->'product'->'metrics');
+
+    -- Update specifications related to the label
+    PERFORM "fertiscan_0.0.11".update_specifications(label_info_id, p_input_json->'specifications');
+
+    -- Update ingredients related to the label
+    PERFORM "fertiscan_0.0.11".update_ingredients(label_info_id, p_input_json->'ingredients');
+
+    -- Update micronutrients related to the label
+    PERFORM "fertiscan_0.0.11".update_micronutrients(label_info_id, p_input_json->'micronutrients');
+
+    -- Update guaranteed analysis related to the label
+    PERFORM "fertiscan_0.0.11".update_guaranteed(label_info_id, p_input_json->'guaranteed_analysis');
+
+    -- Update sub labels related to the label
+    PERFORM "fertiscan_0.0.11".update_sub_labels(label_info_id, p_input_json->'sub_labels');
+
+    -- Update the inspection record
+    verified := (p_input_json->'product'->>'verified')::boolean;
+    inspection_id := "fertiscan_0.0.11".upsert_inspection(
+        p_inspection_id,
+        label_info_id,
+        p_inspector_id,
+        COALESCE(p_input_json->>'sample_id', NULL)::uuid,
+        COALESCE(p_input_json->>'picture_set_id', NULL)::uuid,
+        verified
+    );
+
+    -- Update the inspection ID in the output JSON
+    updated_json := jsonb_set(updated_json, '{inspection_id}', to_jsonb(inspection_id));
+
+    -- Check if the verified field is true, and upsert fertilizer if so
+    IF verified THEN
```

Comment:
Could be a trigger, but this work

Reply:
I will look into how to implement a proper trigger for this.